eduNEXT/platform-plugin-ontask
# feat: ontask openedx connection

### Description
This PR adds an integration for loading course completion data into an OnTask table.

### How to Test
Using Tutor:

1. Install this branch in your environment.
2. Install the Tutor plugin of [OnTask](https://github.com/eduNEXT/tutor-contrib-ontask/?tab=readme-ov-file#installation).
3. Add this inline plugin in your Tutor environment. This allows the OnTask tab to be added to the instructor dashboard and enables the option to add custom settings to a course from the advanced settings in Studio.

    ```yml
    name: ontask-settings
    version: 0.1.0
    patches:
      openedx-common-settings: |
        OPEN_EDX_FILTERS_CONFIG = {
            "org.openedx.learning.instructor.dashboard.render.started.v1": {
                "fail_silently": False,
                "pipeline": [
                    "platform_plugin_ontask.extensions.filters.AddInstructorOnTaskTab",
                ]
            },
        }
      common-env-features: |
        ENABLE_OTHER_COURSE_SETTINGS: true
    ```
4. Go to the instructor dashboard in a course from **LMS**  > **Instructor** > **OnTask**.
5. You will see a **Create Workflow** button to create an Ontask workflow. Click on it, it should show you an error message indicating that the api auth token must be added to the advanced settings of the course.
6. Go to the Advanced Settings from **Studio** > **Settings** > **Advanced Settings** > **Other Course Settings** and add a new key. You can generate the api auth token from **OnTask** > **Profile** >  

    ```json
    {
        ...
        "ONTASK_API_AUTH_TOKEN": "your-ontask-api-auth-token",
    }
    ```
7. Press the button again, an OnTask workflow will be created and redirected to it.
8. Now, click on the **Load Data** button, this will load all the completion information for each student in the course. This operation is executed asynchronously.
9. Finally, go to **Table** > **View data**, you should see all the data loaded.
